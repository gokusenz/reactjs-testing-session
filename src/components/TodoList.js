import React from 'react'

const TodoList = props => (
  <ul className="todo-list" />
)

export default TodoList
