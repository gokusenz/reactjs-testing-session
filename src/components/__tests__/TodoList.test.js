import React from 'react'
import { shallow } from 'enzyme'
import TodoList from '../TodoList'

describe('TodoList', () => {
  let component
  let props

  beforeEach(() => {
    props = {
      todos: [
        { id: 1, text: 'Feed cat', complete: false },
        { id: 2, text: 'Drink coffee', complete: true },
      ],
    }
    component = shallow(<TodoList {...props} />)
  })

  it('renders correct structure', () => {
    expect(component.is('ul')).toBe(true)
    expect(component.hasClass('todo-list')).toBe(true)
  })
})
