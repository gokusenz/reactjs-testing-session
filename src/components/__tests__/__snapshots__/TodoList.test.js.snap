// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList matches its snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="todo-list"
>
    <TodoItem
        complete={false}
        id={1}
        text="Feed cat"
    />
    <TodoItem
        complete={true}
        id={2}
        text="Drink coffee"
    />
</ul>,
  "nodes": Array [
    <ul
      className="todo-list"
>
      <TodoItem
            complete={false}
            id={1}
            text="Feed cat"
      />
      <TodoItem
            complete={true}
            id={2}
            text="Drink coffee"
      />
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TodoList
        todos={
                Array [
                        Object {
                          "complete": false,
                          "id": 1,
                          "text": "Feed cat",
                        },
                        Object {
                          "complete": true,
                          "id": 2,
                          "text": "Drink coffee",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "todos": Array [
            Object {
              "complete": false,
              "id": 1,
              "text": "Feed cat",
            },
            Object {
              "complete": true,
              "id": 2,
              "text": "Drink coffee",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="todo-list"
>
          <TodoItem
                    complete={false}
                    id={1}
                    text="Feed cat"
          />
          <TodoItem
                    complete={true}
                    id={2}
                    text="Drink coffee"
          />
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="todo-list"
>
          <TodoItem
                    complete={false}
                    id={1}
                    text="Feed cat"
          />
          <TodoItem
                    complete={true}
                    id={2}
                    text="Drink coffee"
          />
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoList
    todos={
        Array [
            Object {
              "complete": false,
              "id": 1,
              "text": "Feed cat",
            },
            Object {
              "complete": true,
              "id": 2,
              "text": "Drink coffee",
            },
          ]
    }
/>,
}
`;
